src/Microsoft.AspNetCore.Routing/Tree/TreeRouter.cs

```diff
@@ -192,13 +192,14 @@ public VirtualPathData GetVirtualPath(VirtualPathContext context)
                     {
                         var entry = item.Entry;
                         var matcher = item.TemplateMatcher;
-                        if (!matcher.TryMatch(context.HttpContext.Request.Path, context.RouteData.Values))
-                        {
-                            continue;
-                        }
 
                         try
                         {
+                            if (!matcher.TryMatch(context.HttpContext.Request.Path, context.RouteData.Values))
```

Comment:
It's not throwing any exception - having it in the `try` block just means that `finally` will run even when it invokes `continue;` on line 200.